website_event_membership_restriction/views/event_templates.xml

```diff
@@ -0,0 +1,133 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2024 Tecnativa - Pilar Vargas
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <template
+        id="registration_template"
+        inherit_id="website_event.registration_template"
+        priority="999"
+    >
+        <xpath
+            expr="//form//div[@id='o_wevent_tickets_collapse']//button"
+            position="attributes"
+        >
+            <attribute name="class">d-none</attribute>
+        </xpath>
+        <xpath
+            expr="//form//div[@id='o_wevent_tickets_collapse']//button"
+            position="after"
+        >
+            <button
+                type="submit"
+                t-att-class="'btn btn-primary o_wait_lazy_js a-submit ' + (event.allow_members_only and (not request.env.user or request.env.user.partner_id.commercial_partner_id.membership_state not in ['paid', 'free']) and 'disabled' or '')"
```

Comment:
- Maybe you should check if the user isn't public better `not request.env.user._is_public()`
- Why checking the commercial_partner_id state? What if membership is given by an associaate?

Reply:
@chienandalu do you mean a prorated membership?

Reply:
No, I mean that a contact could be assoiciated to other one membership: https://github.com/odoo/odoo/blob/5fa953a0886c0b45169fc894c6202ede2f68f700/addons/membership/models/partner.py#L13-L15

So a commercial partner could have some child contacts which inherit his membership and others that don't. Also contacts inheriting membership from other organization...

Reply:
That's not the use case, but let's fallback to check if the commercial partner or the partner itself.